tests/testTriggers.go

```diff
@@ -6458,6 +6459,125 @@ func TriggerVolumeUpdate(contexts *[]*scheduler.Context, recordChan *chan *Event
 	updateMetrics(*event)
 }
 
+//TriggerPowerOffVMs
+func TriggerPowerOffAllVMs(contexts *[]*scheduler.Context, recordChan *chan *EventRecord){
+	defer ginkgo.GinkgoRecover()
+	defer endLongevityTest()
+	startLongevityTest(PowerOffAllVMs)
+	event := &EventRecord{
+		Event: Event{
+			ID:   GenerateUUID(),
+			Type: PowerOffAllVMs,
+		},
+		Start:   time.Now().Format(time.RFC1123),
+		Outcome: []error{},
+	}
+	defer func() {
+		event.End = time.Now().Format(time.RFC1123)
+		*recordChan <- event
+	}()
+	setMetrics(*event)
+	stepLog := "Power off all worker nodes test"
+	Step(stepLog, func() {
+		log.Infof(stepLog)
+		workerNodes:= node.GetWorkerNodes()
+		log.Infof("All worker nodes : %v", workerNodes)
+		var numberOfThread int = 5
+		var numberOfNodePerThread int
+		if len(workerNodes) < numberOfThread{
+			numberOfThread = len(workerNodes)
+			numberOfNodePerThread = 1
+		}else{
+			numberOfNodePerThread = len(workerNodes)/numberOfThread
+		}
+		var counter int = 0
+		nodesInThread := make([][]node.Node, numberOfThread)
+		for t := 0; t < numberOfThread;  t++ {
+			nodesInThread[t] = make([]node.Node, numberOfNodePerThread)
+			for n:=0; n < numberOfNodePerThread; n++{
+					nodesInThread[t][n] = workerNodes[counter]
+					counter++
+			}
+		}
+		log.Infof("before additional nodes  : %v", nodesInThread)
+		log.Infof("before additional nodes len  : %v", len(nodesInThread))
+		//Create an additional thread for remainder.
+		if counter < len(workerNodes){
+			log.Infof("Additional nodes  : %d", len(workerNodes) - counter)
+			additonalThread := make([]node.Node, len(workerNodes) - counter)
+			var index int = 0
+			for counter< len(workerNodes) {
+				additonalThread[index] = workerNodes[counter]
+				index++
+				counter++
+			}
+			nodesInThread = append(nodesInThread, additonalThread)
+			numberOfThread++
+		}
+		log.Infof("Complete nodes  : %v", nodesInThread)
+		log.Infof("Complete nodes len  : %v", len(nodesInThread))
+		stepLog = "Power off all worker nodes"
+		Step(stepLog, func() {
+			var poweroffwg sync.WaitGroup
+			for i := 0; i < numberOfThread; i++ {
+				poweroffwg.Add(1)
+				go func(nodeList []node.Node) {
```

Comment:
Is the test case designed to poweroff multiple VMs at once?

Reply:
> Is the test case designed to poweroff multiple VMs at once?

yes